app/scripts/directives/createSecret.js

```diff
@@ -0,0 +1,182 @@
+              if (JSON.parse(data.dockerConfig).auths) {
+                secret.data[".dockerconfigjson"] = encodedConfig;
+              } else {
+                secret.type = "kubernetes.io/dockercfg";
+                secret.data[".dockercfg"] = encodedConfig;
+              }
+              break;
+            case "kubernetes.io/dockercfg":
+              var auth = window.btoa(data.dockerUsername + ":" + data.dockerPassword);
+              var configData = {};
+              configData[data.dockerServer] = {
+                username: data.dockerUsername,
+                password: data.dockerPassword,
+                email: data.dockerMail,
+                auth: auth
+              };
+              secret.data[".dockercfg"] = window.btoa(JSON.stringify(configData));
+              break;
+          }
+          return secret;
+        };
+
+        var linkSecretToServiceAccount = function(secret) {
+          var updatedSA = angular.copy($scope.serviceAccounts[$scope.newSecret.pickedServiceAccountToLink]);
+          if ($scope.newSecret.linkAs.secrets) {
+            updatedSA.secrets.push({name: secret.metadata.name});
+          }
+          if ($scope.newSecret.linkAs.imagePullSecrets) {
+            updatedSA.imagePullSecrets.push({name: secret.metadata.name});
+          }
+          DataService.update('serviceaccounts', $scope.newSecret.pickedServiceAccountToLink, updatedSA, {namespace: $scope.namespace}).then(function(sa) {
+            var alert = {
+              createAndLink: {
+                type: "success",
+                message: "Secret " + secret.metadata.name + " was created and linked with service account " + sa.metadata.name + "."
+              }
+            };
+            $scope.postCreateAction({newSecret: secret, creationAlert: alert});
+          }, function(result){
+            $scope.alerts["createAndLink"] = {
+              type: "error",
+              message: "An error occurred while linking the secret with service account.",
+              details: $filter('getErrorDetails')(result)
+            };
+          });
+        };
+
+        $scope.create = function() {
+          var newSecret = constructSecretObject($scope.newSecret.data, $scope.newSecret.authType);
+          DataService.create('secrets', null, newSecret, $scope).then(function(secret) { // Success
+            if ($scope.newSecret.pickedServiceAccountToLink) {
+              linkSecretToServiceAccount(secret);
+            } else {
+              var alert = {
+                create: {
+                  type: "success",
+                  message: "Secret " + newSecret.metadata.name + " was created."
+                }
+              };
+              $scope.postCreateAction({newSecret: secret, creationAlert: alert});
+            }
+          }, function(result) { // Failure
+            var data = result.data || {};
+            if (data.reason === 'AlreadyExists') {
+              $scope.nameTaken = true;
```

Comment:
I'd return here as a special case. I think the error under the field is enough without the alert.